### DISCUSSION TITLE
_fallback_description.antlers.html optimization

### DISCUSSION THREAD

**Discussion by freshface** [studio1902/statamic-peak#276] [Ideas]
Hi Rob 

In https://github.com/studio1902/statamic-peak/blob/main/dev/resources/views/snippets/_fallback_description.antlers.html#L14
I added a `where="type:article" ` to only target articles.
Because of the `limit="1"` if an article is not the first item of a page builder it will never be reached.

Also there is a check:

```
 {{ if article }}
    {{ article | raw | where('type', 'paragraph') | bard_text | trim | safe_truncate(157, '...') }}
{{ elseif text }}
    {{ text | trim | safe_truncate(157, '...') }}
{{ /if }}
```

There doens't seem to be a type `text`, so i think this is redundant.

New code:

```
{{ seo:collection_defaults | where('collection', {collection}) }}
    {{ if fallback == 'field' }}
        {{ scope:field }}
            {{ field[field_handle] | strip_tags | trim | safe_truncate(157, '...') }}
        {{ /scope:field }}
    {{ elseif fallback == 'custom_text' }}
        {{ custom_text }}
    {{ elseif fallback == 'page_builder' }}
        {{ page_builder where="type:article" limit="1" }}
            {{ article | raw | where('type', 'paragraph') | bard_text | trim | safe_truncate(157, '...') }}
        {{ /page_builder }}
    {{ /if }}
{{ /seo:collection_defaults }}

```

Now we only get the article types.

**Comment by robdekort** [comment]
Hmm. What it currently does is take the first item in the page builder and then see if there's an article in it or a field called `text` (which is what I often use as a fieldname). So it's not wrong persé and ideally you'd get text from as high up in entry as possible. In your example it would only fetch articles when it might make a lot more sense to use a field called `text` in the first page builder block. It usually does on my sites.

I think in the end it's a fallback and probably impossible to get right in all use cases.

What do you think?

**Reply by freshface** [reply]
I agree its just a fallback. But filtering on `where="type:article"` makes sure you have a more "correct" result.

**Reply by robdekort** [reply]
It's not more correct in 9/10 use cases on this side because you limit the use to only articles. The option in the CP states "First text in page builder", not "First text in first article". We could combine the two better maybe?

**Reply by robdekort** [reply]
How about something like this instead:

```antlers
{{ elseif fallback == 'page_builder' }}
    {{ if page_builder:0:text }}
        {{ page_builder:0:text | strip_tags | trim | safe_truncate(157, '...') }}
    {{ else }}
        {{ page_builder where="type:article" limit="1" }}
            {{ if article }}
                {{ article | raw | where('type', 'paragraph') | bard_text | trim | safe_truncate(157, '...') }}
            {{ /if }}
        {{ /page_builder }}
    {{ /if }}
{{ /if }}
```

**Reply by freshface** [reply]
👍🏼 should cover more use cases :)

**Reply by robdekort** [reply]
Let's do it!
